optimade/adapters/base.py

```diff
@@ -19,7 +19,7 @@
 and [`StructureResource`][optimade.models.structures.StructureResource]s, respectively.
 """
 import re
-from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+from typing import Any, Callable, Optional, Union
```

Comment:
Protocol types are deprecated as of Python 3.9 and are alias imports from `collections.abc`.
```suggestion
from collections.abc import Callable
from typing import Any, Optional, Union
```

Reply:
I'd rather stick to the pyupgrade rules really, looks like there was a bug in Python <3.9.3 that broke `collections.abc.Callable` (https://github.com/asottile/pyupgrade/issues/677)